lightningd/payalgo.c

```diff
@@ -321,8 +353,10 @@ static void json_pay(struct command *cmd,
 	}
 	pay->maxfeepercent = maxfeepercent;
 
-	pay->tries = 0;
-	pay->sendpay_parent = NULL;
+	pay->getroute_tries = 0;
+	pay->sendpay_tries = 0;
+	pay->fuzz = 0.75;
```

Comment:
Why is default here 75%, vs 5% for getroute?

Reply:
This is not a default, but an initial value.  `pay` respects `maxfeepercent`, and if it gets a route that has too high fee, it will reduce its `fuzz` value in the hope that the next returned route with a lower fuzz will have a lower fee.  This means that to take advantage of higher-than-usual `maxfeepercent` for improved privacy we should start with a large `fuzz` (which has a chance of getting much higher fee than would be typical for lower or 0 fuzz).

The point of route randomization is to avoid low-fee routes, which might be honeypots used by three-letter monitoring agencies, by occassionally jumping to higher-than-usual fee routes.

Reply:
If 75% is a good starting point for `pay`, why it it not a good default for `getroute`?  It it anticipated that we will see significant failures at 75%?  I'm not sure that's true.  These numbers seem made up, which is fine, but they do need a comment on what we'd need to know to set decent defaults in future...

Reply:
Well, my initial idea was to have `getroute` default to 0%, for compatibility with before this patch, where `getroute` never fuzzed and (if there were no changes in network topology) would always return the same route again and again.  A mere 5% for fuzz is unlikely to change anything, either.

If *that* behavior is not necessary for `getroute`, then 75% would be a reasonable default for `getroute`, too.